### ISSUE TITLE
IndexSelect Error

### ISSUE THREAD

**Issue by JorgenWan** [dmlc/dgl#2469]
## To Reproduce

1.build a list of graphs from dgl.graph()
2.batch them use dgl.batch()
3.call model.forward() , a model like GraphConv

## Expected behavior
Forward normally

## Environment

 - DGL Version: 0.5.3
 - Backend Library & Version: PyTorch 1.7.0
 - OS: Ubuntu 16.04
 - How you installed DGL: pip
 - Build command you used: no
 - Python version: 3.6.5
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration: GeForce GTX 1080Ti

## Error Info

> terminate called after throwing an instance of 'dmlc::Error'
  what():  [23:44:14] /opt/dgl/src/runtime/cuda/cuda_device_api.cc:103: Check failed: e == cudaSuccess || e == cudaErrorCudartUnloading: CUDA: an illegal memory access was encountered
Stack trace:
  [bt] (0) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x4f) [0x7f5f4bd00c4f]
  [bt] (1) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(dgl::runtime::CUDADeviceAPI::FreeDataSpace(DLContext, void*)+0x15c) [0x7f5f4c51c77c]
  [bt] (2) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(dgl::runtime::NDArray::Internal::DefaultDeleter(dgl::runtime::NDArray::Container*)+0x1ad) [0x7f5f4c3de4cd]
  [bt] (3) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(dgl::runtime::NDArray dgl::aten::impl::IndexSelect<(DLDeviceType)2, long, long>(dgl::runtime::NDArray, dgl::runtime::NDArray)+0x3a3) [0x7f5f4c525943]
  [bt] (4) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(dgl::aten::IndexSelect(dgl::runtime::NDArray, dgl::runtime::NDArray)+0xb25) [0x7f5f4bcebf35]
  [bt] (5) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(void dgl::aten::impl::COOSort_<(DLDeviceType)2, long>(dgl::aten::COOMatrix*, bool)+0x206) [0x7f5f4c54ef46]
  [bt] (6) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(dgl::aten::COOSort_(dgl::aten::COOMatrix*, bool)+0x344) [0x7f5f4bce5594]
  [bt] (7) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(dgl::aten::COOSort(dgl::aten::COOMatrix, bool)+0x471) [0x7f5f4bd27ca1]
  [bt] (8) /home/578177149/anaconda3/lib/python3.6/site-packages/dgl/libdgl.so(dgl::aten::CSRMatrix dgl::aten::impl::COOToCSR<(DLDeviceType)2, long>(dgl::aten::COOMatrix)+0x5af) [0x7f5f4c54e41f]

**Comment by yzh119** [comment]
Hi @JorgenWan would you mind sharing the graph data you used so that we can locate the bug?

**Comment by JorgenWan** [comment]
The graph data is hard to extract. In my code, I will build a list of graphs, like this

sgs = []
for sg_nodes in sgs_nodes:
    tmp_sg = g.subgraph(sg_nodes)
    sg = dgl.graph((tmp_sg.edges()[0].tolist(), tmp_sg.edges()[1].tolist()),     num_nodes=len(sg_nodes)+1)
    span_node = int(sg.num_nodes() - 1)
    edges_src = [span_node for _ in range(len(span_weights))]
    edges_tgt = [n for n, _ in span_weights]
    sg.add_edges(edges_src, edges_tgt)
    sgs.append(sg)

batched_g = dgl.batch(sgs).to(features.device)
h = features
for i, layer in enumerate(self.layers):
     h = F.dropout(h, p=self.gcn_dropout, training=self.training)
     h = layer(g, h) # where bug start! # layer is GraphConv

**Comment by jermainewang** [comment]
Hi @JorgenWan , since we cannot reproduce the problem, I will close this issue for now. Feel free to reopen if the problem still exists. Thanks!

**Comment by LILI-ZHOU** [comment]
I also met the same mistake, could you solve it?

**Comment by JorgenWan** [comment]
> I also met the same mistake, could you solve it?

I limit the maximal number of edges in each batched graph, it seems that it solves in a degree. But when I use another dataset, it seems I will meet the problem again. It is strange. I don't know whether it is related to the  GPU memory usage of DGL (O(|V|) or O(|E|) or something) .

**Comment by LILI-ZHOU** [comment]
> > I also met the same mistake, could you solve it?
> 
> I limit the maximal number of edges in each batched graph, it seems that it solves in a degree. But when I use another dataset, it seems I will meet the problem again. It is strange. I don't know whether it is related to the GPU memory usage of DGL (O(|V|) or O(|E|) or something) .

It's strange that I met this problem with version 0.5.3.  When I lowered the version to 0.4, the problem no longer occurred, even though I didn't change any code

**Comment by JorgenWan** [comment]
> > > I also met the same mistake, could you solve it?
> > 
> > 
> > I limit the maximal number of edges in each batched graph, it seems that it solves in a degree. But when I use another dataset, it seems I will meet the problem again. It is strange. I don't know whether it is related to the GPU memory usage of DGL (O(|V|) or O(|E|) or something) .
> 
> It's strange that I met this problem with version 0.5.3. When I lowered the version to 0.4, the problem no longer occurred, even though I didn't change any code

emmm, how to uninstall DGL 0.5.3 and install DGL 0.4?  pip install dgl-cu101 will install 0.5.3 directly.

**Comment by JorgenWan** [comment]
Hi, I need some help. I use "pip install dgl-cu101 " and my ubuntu will install DGL 0.5.3.
But I want to install DGL 0.4, how can I unstall my DGL 0.5.3 and install DGL 0.4?
I would appreciate it if you could help!



----- 原始邮件 -----
发件人: "Zihao Ye" <notifications@github.com>
收件人: "dmlc/dgl" <dgl@noreply.github.com>
抄送: "Juncheng Wan" <578177149@sjtu.edu.cn>, "Mention" <mention@noreply.github.com>
发送时间: 星期一, 2021年 2 月 01日 上午 9:35:50
主题: Re: [dmlc/dgl] IndexSelect Error (#2469)

Reopened #2469.

-- 
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub:
https://github.com/dmlc/dgl/issues/2469#event-4271080802

**Comment by VoVAllen** [comment]
`pip install "dgl-cu101<0.5"` should work

**Comment by JorgenWan** [comment]
After installing DGL 0.4, this error no longer occurred.

**Comment by VoVAllen** [comment]
@JorgenWan It still seems a bug of DGL. Did you meet error with multi-gpu program or single-gpu?

**Comment by JorgenWan** [comment]
> @JorgenWan It still seems a bug of DGL. Did you meet error with multi-gpu program or single-gpu?

I meet this error with single-gpu.

**Comment by xjy-xing** [comment]
I meet this error with single-gpu. too . with version  dgl-cu101 0.6.1

**Comment by mufeili** [comment]
> I meet this error with single-gpu. too . with version dgl-cu101 0.6.1

Could you try the latest version of DGL? This is a very old issue thread. Should you have further questions, you may open a new issue. Thanks.
